cmd/dist/snapshot.go

```diff
@@ -0,0 +1,332 @@
+		var (
+			ctx    = contextpkg.Background()
+			root   = context.GlobalString("root")
+			key    = context.Args().First()
+			parent = context.Args().Get(1)
+		)
+
+		root, err := filepath.Abs(root)
+		if err != nil {
+			return err
+		}
+		log.G(ctx).WithField("root", root).Debug("setting up snapshot driver")
+
+		driver, err := overlay.NewSnapshotter(root)
+		if err != nil {
+			return err
+		}
+
+		if key == "" {
+			return fmt.Errorf("key required")
+		}
+
+		mounts, err := driver.Prepare(ctx, key, parent)
+		if err != nil {
+			return err
+		}
+
+		for _, mount := range mounts {
+			fmt.Println(mount)
+		}
+
+		return nil
+	},
+}
+
+var snapshotViewCommand = cli.Command{
+	Name:        "view",
+	Usage:       "Create a view of snapshot",
+	ArgsUsage:   "<key> <parent>",
+	Description: `Creates an View of Snapshot layer [Read Only]`,
+	Action: func(context *cli.Context) error {
+		var (
+			ctx    = contextpkg.Background()
+			root   = context.GlobalString("root")
+			key    = context.Args().First()
+			parent = context.Args().Get(1)
+		)
+
+		root, err := filepath.Abs(root)
+		if err != nil {
+			return err
+		}
+		log.G(ctx).WithField("root", root).Debug("setting up snapshot driver")
+
+		driver, err := overlay.NewSnapshotter(root)
+		if err != nil {
+			return err
+		}
+
+		if key == "" {
+			return fmt.Errorf("key required")
+		}
+
+		mounts, err := driver.View(ctx, key, parent)
+		if err != nil {
+			return err
+		}
+
+		for _, mount := range mounts {
+			fmt.Println(mount)
+		}
+
+		return nil
+	},
+}
+
+var snapshotRemoveCommand = cli.Command{
+	Name:      "remove",
+	Usage:     "Remove a snapshot",
+	ArgsUsage: "<key> ",
+	Action: func(context *cli.Context) error {
+		var (
+			ctx  = contextpkg.Background()
+			root = context.GlobalString("root")
+			key  = context.Args().First()
+		)
+
+		root, err := filepath.Abs(root)
+		if err != nil {
+			return err
+		}
+		log.G(ctx).WithField("root", root).Debug("setting up snapshot driver")
+
+		driver, err := overlay.NewSnapshotter(root)
+		if err != nil {
+			return err
+		}
+
+		if key == "" {
+			cli.ShowSubcommandHelp(context)
+			return fmt.Errorf("key required")
+		}
+
+		err = driver.Remove(ctx, key)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+}
+
+var snapshotCommitCommand = cli.Command{
+	Name:        "commit",
+	Usage:       "commit an active snapshot",
+	ArgsUsage:   "<name <key>",
+	Description: `Commit the Active layer with name provided`,
+	Action: func(context *cli.Context) error {
+		var (
+			ctx  = contextpkg.Background()
+			root = context.GlobalString("root")
+			name = context.Args().First()
+			key  = context.Args().Get(1)
+		)
+
+		root, err := filepath.Abs(root)
+		if err != nil {
+			return err
+		}
+		log.G(ctx).WithField("root", root).Debug("setting up snapshot driver")
+
+		driver, err := overlay.NewSnapshotter(root)
+		if err != nil {
+			return err
+		}
+
+		if name == "" {
+			cli.ShowSubcommandHelp(context)
```

Comment:
cli package show help only with ``--help`` . Since, the multiple arguments are passed as positional  inputs , So showing help will be helpful in case insufficient arguments.